src/main/java/io/github/jroy/happybot/events/star/StarMessages.java

```diff
@@ -26,23 +26,51 @@
 @SuppressWarnings("FieldCanBeLocal")
 public class StarMessages extends ListenerAdapter {
   private static final String NEW_GILDED_MESSAGE = "New Gilded Message";
-  private final String CREATE_TABLE = "CREATE TABLE IF NOT EXISTS starstats ( `id` INT(50) NOT NULL AUTO_INCREMENT , `userid` VARCHAR(50) NOT NULL , `stars` BIGINT(255) NOT NULL DEFAULT '0' , `gilds` BIGINT(255) NOT NULL DEFAULT '0' , `heels` BIGINT(255) NULL DEFAULT '0' , PRIMARY KEY (`id`)) ENGINE = InnoDB;";
+
+  private final String CREATE_STAT_TABLE = "CREATE TABLE IF NOT EXISTS starstats ( `id` INT(50) NOT NULL AUTO_INCREMENT , `userid` VARCHAR(50) NOT NULL , `stars` BIGINT(255) NOT NULL DEFAULT '0' , `gilds` BIGINT(255) NOT NULL DEFAULT '0' , `heels` BIGINT(255) NULL DEFAULT '0' , PRIMARY KEY (`id`)) ENGINE = InnoDB;";
   private final String SELECT_USER = "SELECT * FROM starstats WHERE userid = ?;";
   private final String CREATE_USER = "INSERT INTO starstats (userId) VALUES (?);";
   private final String UPDATE_USER = "UPDATE starstats SET stars = ?, gilds = ?, heels = ? WHERE userId = ?;";
+
+  private final String CREATE_USED_TABLE = "CREATE TABLE IF NOT EXISTS starused ( `id` INT(50) NOT NULL AUTO_INCREMENT , `messageId` VARCHAR(255) NOT NULL , PRIMARY KEY (`id`)) ENGINE = InnoDB;";
+  private final String ADD_USED = "INSERT INTO starused (messageId) VALUES (?);";
+  private final String SELECT_USED = "SELECT * FROM starused WHERE messageId = ?;";
+
   private Connection connection;
   private Set<String> alreadyUsedMessages = new HashSet<>();
   private Map<String, GildInfoToken> pastGilds = new HashMap<>();
 
   public StarMessages(SQLManager sqlManager) {
     connection = sqlManager.getConnection();
     try {
-      connection.createStatement().executeUpdate(CREATE_TABLE);
+      connection.createStatement().executeUpdate(CREATE_STAT_TABLE);
+      connection.createStatement().executeUpdate(CREATE_USED_TABLE);
+    } catch (SQLException e) {
+      e.printStackTrace();
+    }
+  }
+
+  private void addUsed(String messageId) {
```

Comment:
Basic documentation for these internal functions would make the code much easier to follow.